sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -126,25 +146,84 @@
 
   @Override
   public Span startSpan() {
-    // TODO: get remoteParent span from the context if noParent=false and remoteParent/remoteParents
-    // are null
-    // TODO: correctly implement this.
-    SpanContext context =
-        SpanContext.create(
-            TraceId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}, 0),
-            SpanId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8}, 0),
-            TraceOptions.builder().setIsSampled(true).build(),
-            Tracestate.getDefault());
+    Span currentSpan = null;
+    if (parentType == ParentType.CURRENT_SPAN) {
+      currentSpan = ContextUtils.getValue();
+    }
+    SpanContext parentContext = parent(parentType, currentSpan, parent, remoteParent);
+    SpanContext spanContext = createSpanContext(spanName, parentContext, sampler);
+    boolean recordEvents =
+        this.recordEvents != null ? this.recordEvents : spanContext.getTraceOptions().isSampled();
+    TimestampConverter timestampConverter = getTimestampConverter(parent);
     return RecordEventsReadableSpanImpl.startSpan(
```

Comment:
If `recordEvents` is `false`, `DefaultSpan` should be returned, rather than `RecordEventsReadableSpanImpl `.

Reply:
I thought this would only affect `addEvents(` methods. Whereas `DefaultSpan` is not reported at all.

Reply:
`DefaultSpan` does not expose factory method. 

I will add it to todos and do it in a follow up PR.

Reply:
It will be done in https://github.com/open-telemetry/opentelemetry-java/issues/438